module/VuFind/src/VuFind/RecordDriver/WorldCat2.php

```diff
@@ -0,0 +1,551 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+
+namespace VuFind\RecordDriver;
+
+use function count;
+
+/**
+ * Model for WorldCat v2 records.
+ *
+ * @category VuFind
+ * @package  RecordDrivers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+class WorldCat2 extends DefaultRecord
+{
+    /**
+     * Return the unique identifier of this record within the index;
+     * useful for retrieving additional information (like tags and user
+     * comments) from the external MySQL database.
+     *
+     * @return string Unique identifier.
+     */
+    public function getUniqueID()
+    {
+        if (!isset($this->fields['identifier']['oclcNumber'])) {
+            throw new \Exception('ID not set!');
+        }
+        return $this->fields['identifier']['oclcNumber'];
+    }
+
+    /**
+     * Get the call numbers associated with the record (empty array if none).
+     *
+     * @return array
+     */
+    public function getCallNumbers()
+    {
+        $retVal = [];
+        foreach (['lc', 'dewey'] as $type) {
+            $retVal = array_merge($retVal, (array)($this->fields['classification'][$type] ?? []));
+        }
+        return $retVal;
+    }
+
+    /**
+     * Get the Dewey call number associated with this record (empty string if none).
+     *
+     * @return string
+     */
+    public function getDeweyCallNumber()
+    {
+        return ((array)($this->fields['classification']['dewey'] ?? []))[0] ?? '';
+    }
+
+    /**
+     * Get a raw, unnormalized LCCN. (See DefaultRecord::getLCCN for normalization).
+     *
+     * @return string
+     */
+    protected function getRawLCCN()
+    {
+        return ((array)($this->fields['classification']['lc'] ?? []))[0] ?? '';
+    }
+
+    /**
+     * Get an array of all the formats associated with the record.
+     *
+     * @return array
+     */
+    public function getFormats()
+    {
+        $formats = [];
+        foreach (['generalFormat', 'specificFormat'] as $key) {
+            if (isset($this->fields['format'][$key])) {
+                $formats[] = $this->translate(
+                    'WorldCatFormats::' . strtolower($this->fields['format'][$key]),
```

Comment:
Agreed. In the long term, I'd like to do a more general overhaul of format handling -- move the core format names out of the default text domain and into a format-specific one, and use aliasing instead of in-code conversion to map equivalent formats from various backends. But alas, it's a lot of work, and not a top priority, so it hasn't gotten done yet. For now, my goal here was to have reasonable support for the top-level WorldCat formats, but I didn't attempt to address all of the sub-categories. If there's sufficient interest, I am of course happy to, but I didn't want to create tons of work for the translators if nobody cared. :-)

Reply:
Yeah I think what is done in this PR is certainly sufficient, and as you say doesn't create work for the translators.  A broader review sometime in the future would make sense.